Game/Source/GDKTestGyms/BenchmarkGymGameModeBase.cpp

```diff
@@ -307,20 +350,55 @@ void ABenchmarkGymGameModeBase::TickUXMetricCheck(float DeltaSeconds)
 
 	if (!bHasUxFailed &&
 		!bUXMetricValid &&
-		Constants->UXMetricDelay.IsReady())
+		Constants->UXMetricDelay.HasTimerGoneOff())
 	{
 		bHasUxFailed = true;
-		NFR_LOG(LogBenchmarkGymGameModeBase, Error, TEXT("UX metric has failed. RTT: %.8f, UpdateDelta: %.8f, ActivePlayers: %d"), AveragedClientRTTSeconds, AveragedClientUpdateTimeDeltaSeconds, ActivePlayers);
+		NFR_LOG(LogBenchmarkGymGameModeBase, Error, TEXT("NFR scenario failed: UX metric check. RTT: %.8f, UpdateDelta: %.8f, ActivePlayers: %d"), AveragedClientRTTSeconds, AveragedClientUpdateTimeDeltaSeconds, ActivePlayers);
 	}
 
-	PrintUXMetricTimer -= DeltaSeconds;
-	if (PrintUXMetricTimer < 0.0f)
+	if (PrintMetricsTimer.HasTimerGoneOff())
 	{
-		PrintUXMetricTimer = 10.0f;
 		NFR_LOG(LogBenchmarkGymGameModeBase, Log, TEXT("UX metric values. RTT: %.8f(%d), UpdateDelta: %.8f(%d), ActivePlayers: %d"), AveragedClientRTTSeconds, ValidRTTCount, AveragedClientUpdateTimeDeltaSeconds, ValidUpdateTimeDeltaCount, ActivePlayers);
 	}
 }
 
+void ABenchmarkGymGameModeBase::TickActorCountCheck(float DeltaSeconds)
+{
+	const UNFRConstants* Constants = UNFRConstants::Get(GetWorld());
+	check(Constants);
+
+	// This test respects the initial delay timer in both native and GDK
+	if (!bHasActorCountFailed &&
```

Comment:
I thought we agreed to do this all the time (ie. set bHasActorCountFailed) but only actually fail the scenario when the timer has expired?

Reply:
We still get the print out of how many actors are in the scenario as that is done by the counter component on the game state. The only purpose of further logic is to print the failure. Unless I am missing something?

Reply:
It's also to register the metric back to grafana - which won't be done - or it will be done but with a potentially "fake" value.

Reply:
Ahh yes! Will change.